contracts/Removal.sol

```diff
@@ -119,18 +200,33 @@ contract Removal is
     }
   }
 
+  function mint(
+    address,
+    uint256,
+    uint256,
+    bytes calldata
+  ) public pure override {
+    revert("Removal: ERC 1155 mint disabled");
+  }
+
   /**
-   * @dev used to initiate a sale of removals by transferring the removals to the market contract
+   * @dev used to list removals for sale by transferring the removals to the market contract
+   *
+   * ### Requirements:
+   *  - all removals being listed must belong to the same project id.
    */
   function safeBatchTransferFrom(
     address _from,
     address _to,
-    uint256[] memory _ids,
+    uint256[] memory ids,
     uint256[] memory _amounts,
     bytes memory _data
   ) public override {
+    // todo do we add any validation to enforce that all removals in batch belong to the same project id?
+    uint256 projectId = _removalIdToProjectId[ids[0]];
+    _restrictedNori.createSchedule(projectId);
```

Comment:
this seems like it would be more appropriately housed in the rNORI logic

Reply:
well the `createSchedule` logic is in `rNORI`... this Removal contract has to call _something_ to set up the correct schedule for the batch, so I'm not clear on what change you'd be proposing here

Reply:
perhaps it should live in `onERC1155BatchReceived` in the market contract

Reply:
I believe that would also fix [this](https://github.com/nori-dot-eco/contracts/security/code-scanning/485) `check-effects-interaction` violation

Reply:
pretty sure the interface check happens before `onERC1155BatchReceived` is called so not sure how it could bubble up something it hasn't reached yet